[9.0] [MIG] base_location_geonames_import

pedrobaeza opened the pull request:
- Make module installable
- Short headers
- Update README.rst
- Improve test efficiency

pedrobaeza committed:
[MIG] base_location_geonames_import: Initial migration to 9.0

pedrobaeza committed:
[IMP] base_location_geonames_import: Make tests more efficient

bealdav commented:
FYI, yml will be deprecated

https://www.odoo.com/fr_FR/groups/community-59/community-14056914

pedrobaeza commented:
Thanks for the info, and I prefer Python files, but for this module, I take the easy path of only update existing YAML file. As in v9 is still supported, I prefer to let as is.

lmignon commented on base_location_geonames_import/wizard/geonames_import.py:
```diff
@@ -1,29 +1,11 @@
 
 from openerp import models, fields, api, _
-from openerp.exceptions import Warning
+from openerp.exceptions import Warning as UserError
```
Why do you have aliased 'Warning'?

pedrobaeza replied:
This is a guideline because Warning is deprecated

pedrobaeza replied:
https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#idioms

pedrobaeza replied:
I'm seeing that in v9 I should import directly UserError

pedrobaeza replied:
@lmignon, changed to use directly `UserError`

pedrobaeza committed:
[FIX] base_location_geonames_import: Don't alias Warning

lmignon commented:
Thank you @pedrobaeza the code looks good to me. :+1: once Travis is green.

pedrobaeza committed:
[IMP] base_location_geonames_import: Convert tests to Python + switch imported country

Monaco country is very little and it allows to save some downloaded bytes plus making
a test for entries deletion

pedrobaeza commented:
@bealdav, with the advices from @alexis-via in https://github.com/OCA/partner-contact/pull/254, I have converted tests to Python, and improve a bit more the efficiency.

simahawk commented:
:+1:

pedrobaeza commented:
We can merge this one.
